src/ApiService/ApiService/onefuzzlib/Config.cs

```diff
@@ -257,4 +264,257 @@ await _containers.GetContainerSasUrl(x.Item2.Name, StorageType.Corpus, ConvertPe
+        }
+
+        if (config.Pool == null) {
+            return ResultVoid<TaskConfigError>.Error(new TaskConfigError("pool must be specified"));
+        }
+
+        if (!CheckVal(definition.Vm.Compare, definition.Vm.Value, config.Pool!.Count)) {
+            var err =
+                $"invalid vm count: expected {definition.Vm.Compare} {definition.Vm.Value}, got {config.Pool.Count}";
+            _logTracer.Error(err);
+            return ResultVoid<TaskConfigError>.Error(new TaskConfigError(err));
+        }
+
+        var pool = await _context.PoolOperations.GetByName(config.Pool.PoolName);
+        if (!pool.IsOk) {
+            return ResultVoid<TaskConfigError>.Error(new TaskConfigError($"invalid pool: {config.Pool.PoolName}"));
+        }
+
+        var checkTarget = await CheckTargetExe(config, definition);
+        if (!checkTarget.IsOk) {
+            return checkTarget;
+        }
+
+        if (definition.Features.Contains(TaskFeature.GeneratorExe)) {
+            var container = config.Containers!.First(x => x.Type == ContainerType.Tools);
+
+            if (config.Task.GeneratorExe == null) {
+                return ResultVoid<TaskConfigError>.Error(new TaskConfigError($"generator_exe is not defined"));
+            }
+
+            var tool_paths = new[] { "{tools_dir}/", "{tools_dir}\\" };
+
+            foreach (var toolPath in tool_paths) {
+                if (config.Task.GeneratorExe.StartsWith(toolPath)) {
+                    var generator = config.Task.GeneratorExe.Replace(toolPath, "");
+                    if (!await _containers.BlobExists(container.Name, generator, StorageType.Corpus)) {
+                        var err =
+                            $"generator_exe `{config.Task.GeneratorExe}` does not exist in the tools container `{container.Name}`";
+                        _logTracer.Error(err);
+                        return ResultVoid<TaskConfigError>.Error(new TaskConfigError(err));
+                    }
+                }
+            }
+        }
+
+        if (definition.Features.Contains(TaskFeature.StatsFile)) {
+            if (config.Task.StatsFile != null && config.Task.StatsFormat == null) {
+                var err2 = "using a stats_file requires a stats_format";
+                _logTracer.Error(err2);
+                return ResultVoid<TaskConfigError>.Error(new TaskConfigError(err2));
+            }
+        }
+
+        return ResultVoid<TaskConfigError>.Ok();
+
+    }
+
+    private async Task<ResultVoid<TaskConfigError>> CheckTargetExe(TaskConfig config, TaskDefinition definition) {
+        if (config.Task.TargetExe == null) {
+            if (definition.Features.Contains(TaskFeature.TargetExe)) {
+                return ResultVoid<TaskConfigError>.Error(new TaskConfigError("missing target_exe"));
+            }
+
+            if (definition.Features.Contains(TaskFeature.TargetExeOptional)) {
+                return ResultVoid<TaskConfigError>.Ok();
```

Comment:
is there value to have this "if" statement ? the return from the function is the same as from within the if statement

Reply:
yeah i thought it was weird too but I kept it to match the [original code ](https://github.com/chkeita/onefuzz/blob/b2399c45710b52160b994e429fe3393a292da8be/src/api-service/__app__/onefuzzlib/tasks/config.py#L109)